### ISSUE TITLE
location and orientation of split violins

### ISSUE THREAD

**Issue by AdityaAsopa** [mwaskom/seaborn#3619]
Thanks for the amazing work btw.

My problem is about placement of violinplots. I only need the left half of the violins for my categories.

`sns.violinplot(data=df_temp, x="numSq", y="cell_fpr", hue="numSq", dodge=True,  inner=None, split=True, ax=ax1d, native_scale=False)`

I am using split=True 

but seaborn is deciding two things without my control: 1) showing either right or left half of the split. 2) setting the position of the violins at arbitrary positions on the x-axis.
![Figure 15](https://github.com/mwaskom/seaborn/assets/11498989/214e54d3-6b09-407d-89c9-2bb93fd9e652)

split=False gives full violins at correct places.
![Figure 16](https://github.com/mwaskom/seaborn/assets/11498989/08b8e71a-d393-4938-aa73-8d15a7318dd5)

Using dodge=True only increases spacing between the axis ticks
![Figure 18](https://github.com/mwaskom/seaborn/assets/11498989/e9e4385f-4d84-4323-8e90-c6655ec8058a)

Is there a way to control which half of the violin to show and more importantly, the location of the violins w.r.t x-ticks. Many thanks

**Comment by thuiop** [comment]
Hello !
The alternating left-right distribution is due to the fact that you use the hue parameter ; it works that way in order to produce "face to face" violin plots, such as this one (taken from the documentation). If you really want your split violins, you should do without the hue. The violins are also not placed at random locations, they are placed such that they are centered around the category name. Otherwise, you would have a lot of empty space. ![violinplot_9_0](https://github.com/mwaskom/seaborn/assets/1338337/17136a29-edf3-402a-aa39-8f365700c85a)
As a side note, I personally don't see the use case for this. If you actually want these plain one-sided violins, why not use a KDE or an histogram in the first place ? If this is because of many categories, the "ridge" plot may be suited to your needs https://seaborn.pydata.org/examples/kde_ridgeplot.html. (this is only a suggestion, feel free to use whatever suits you the most).
Finally, it is best to provide a reproducible example when you open an issue, preferably using one of the seaborn built-in datasets.
Hope this helps.

**Comment by AdityaAsopa** [comment]
Thanks for your work and reply. I take it that there is currently no way to control the orientation and placement of the half violins. I will surely check out the ridge/KDE plots, those also look close to what I had in mind (a compound plot of half violin half swarm, something like a raincloud plot)

Please close the ticket if you wish. I think it would help if one could get the position and orientation control in violinplots also though. Cheers

**Comment by mwaskom** [comment]
Thanks @thuiop that's a good summary. `split` has grown in flexibility a bit recently, but the underlying code still works mostly assuming that you're using it to compare between hue values at each location. And agreed that the locations are not random. I'm probably -1 on any additional API expansion to support further customization of split violin plots; I think we'd need several additional parameters to support exactly what you're trying to do and they would only make sense (or would work differently) in the context of `split=True`, which is already a sub-feature.

Note that you may be able to get close to what you want using `hue_order`, e.g.

```python
sns.violinplot(tips, x="day", y="total_bill", split=True, hue=1, hue_order=[1, 2], dodge=True, legend=False)
sns.stripplot(tips, x="day", y="total_bill", hue=1, hue_order=[2, 1], dodge=True, legend=False)
```
![image](https://github.com/mwaskom/seaborn/assets/315810/0bbaacd1-afed-4921-9e0e-b6f4e6277b7c)

**Comment by AdityaAsopa** [comment]
This is a very neat solution and well within what seaborn can already do. Feels like I should have tried a little more.
Danke schön

**Comment by Mitmischer** [comment]
@thuiop @mwaskom I'm sorry to ping you directly but as the issue is closed, this is my only way to get attention.

I have an issue with this, too.
But I do not use the hue parameter.

Consider the following example:
```
pd.DataFrame({"young":[11,14,30,20,24], "old":[10,29,17,53,27]})
sns.violinplot(pd.DataFrame(ep_num_dfs["ttr"][0:5], columns=["young", "old"]), fill=True, inner="box", split=True, orient="h")
```

That will give me the following output:
![image](https://github.com/user-attachments/assets/fc7bd0b8-5a42-453a-9a89-2aabf12af9e8)

.. so it does the side-by-side-splits although I'm not using hue. If I apply your suggestion from above, if will even tell me:

```
ValueError: The following variable cannot be assigned with wide-form data: `hue`
```

Do I have to convert my data frame to long form for this to work?. But shouldn't this issue here be independent from the data-frame layout? Also, my data frame isn't really in a wide format either, I abuse it just as a collection of columns. So I don't really know how I'd properly convert it to long format anyway.
In summary, how can I make seaborn use the same side for my densities without tinkering with hue?

**Comment by thuiop** [comment]
Yes, this is an issue be cause it is a wide-format thing. I am not sure how to explain how to convert to long-form since I don't know what your data looks like, but basically you should have a 10 row dataframe with a column "age" with 5 rows being "young" and 5 rows being "old" for instance. This may help https://pandas.pydata.org/docs/reference/api/pandas.melt.html

**Comment by Mitmischer** [comment]
@thuiop Cool! That example cleared it up for me (I can just use a column "column" and use the column name in there). I'd find it more convenient if the documentation in the wide-format case here was a bit clearer (as hue is at least apparently not involved) but my problem is solved and I thank you a lot for that!
